### PR TITLE
Add Escape Valve for check test results the old way

### PR DESCRIPTION
**Pull request by alexperovich**
@RussKeldorph this will enable the old way to check results as an opt-in.

### CODE REVIEW

File: tests/UnitTests.proj [hunk 1]
```diff
@@ -26,8 +28,8 @@
     </HelixWorkItem>
   </ItemGroup>
   <ItemGroup>
-    <AzurePipelinesExpectedTestFailure Include="PizzaTests.cheese"/>
-    <AzurePipelinesExpectedTestFailure Include="PizzaTests.cheese2"/>
+    <!-- <AzurePipelinesExpectedTestFailure Include="PizzaTests.cheese"/>
```
Comment (ChadNedzlek) [inline tests/UnitTests.proj]: ? Commented out?

Comment (alexperovich) [reply tests/UnitTests.proj]: Yea, I am gona revert this commit once I verify the change. This is to verify the thing I wrote will work in azdev.

**Comment by RussKeldorph** [comment]
@BruceForstall @echesakovMSFT

**Comment by RussKeldorph** [comment]
I'm not extremely familiar with this issue, but, after talking to @BruceForstall and @echesakovMSFT, this new behavior of Azure Pipelines sounds broken.  Are we tracking getting the AzP team to fix it?

@alexperovich @MattGal @chcosta

**Comment by alexperovich** [comment]
I have a conversation ongoing with them. The problem is the api they pointed me too isn't run specific. If there isn't a run specific api then we will have to explore other options.

**Comment by echesakovMSFT** [comment]
@alexperovich I don't know much about the apis AzDO has but I am wondering whether it is possible to use the two apis:
1) the old one to get all the test failures for given test run id;
2) the new one for filtering out the flaky test based on the knowledge of all the failed non-flaky tests in a build;
This could be a kind of intersection of the results of both queries?

**Comment by alexperovich** [comment]
The new api is nothing more than the counts of test results per status. There isn't any documentation for it either so I don't know if there are other apis in the same set that I can use to get results for specific runs.

**Pull request merged**